lib/DependencyScan/ModuleDependencyScanner.cpp

```diff
@@ -153,6 +153,17 @@ ModuleDependencyScanningWorker::ModuleDependencyScanningWorker(
   auto ClangModuleCachePath = getModuleCachePathFromClang(
       ScanASTContext.getClangModuleLoader()->getClangInstance());
   auto &FEOpts = ScanCompilerInvocation.getFrontendOptions();
+
+  // Configure the filesystem to use the same shared `stat` cache as the Clang
+  // worker uses.
+  llvm::IntrusiveRefCntPtr<
+      clang::tooling::dependencies::DependencyScanningWorkerFilesystem>
+      DepFS =
+          new clang::tooling::dependencies::DependencyScanningWorkerFilesystem(
+              globalScanningService.ClangScanningService->getSharedCache(),
+              ScanASTContext.SourceMgr.getFileSystem());
+  ScanASTContext.SourceMgr.setFileSystem(std::move(DepFS));
```

Comment:
What's the lifetime of `ScanASTContext`?

I'm asking because each instance of `DependencyScanningWorkerFilesystem` has a copy of a subset of the shared cache to reduce lock contention in the shared cache. We should create as few of these VFSs as possible to keep things efficient.

Reply:
Its lifetime is the entire scan. And until we enable parallel-scanning by-default, there is only one of these workers.